logger.go

```diff
@@ -38,30 +38,42 @@ func (l *LevelLogger) EnableDebug(enable bool) {
 	l.debug = enable
 }
 
-func (l LevelLogger) DebugPrintf(format string, a ...interface{}) {
-	if l.debug {
-		l.Print("[DEBUG] ")
-		l.Printf(format, a...)
-	}
+func (l *LevelLogger) EnableTrace(enable bool) {
+	l.trace = enable
 }
 
-func (l LevelLogger) TracePrintf(format string, a ...interface{}) {
+func (l LevelLogger) DebugPrintf(format string, a ...interface{}) {
 	if l.debug {
-		l.Print("[TRACE] ")
-		l.Printf(format, a...)
+		l.Printf("[DEBUG] " + format, a...)
 	}
 }
 
 func (l LevelLogger) DebugPrint(a ...interface{}) {
 	if l.debug {
-		l.Print("[DEBUG] ")
-		l.Print(a...)
+		l.Print(append([]interface{}{"[DEBUG] "}, a))
 	}
 }
 
 func (l LevelLogger) DebugPrintln(a ...interface{}) {
 	if l.debug {
-		l.Print("[DEBUG] ")
-		l.Println(a...)
+		l.Println(append([]interface{}{"[DEBUG] "}, a))
```

Comment:
forgot to flatten  args and the new slice

https://play.golang.org/p/XgE0M24oL15

````go
package main

import (
	"log"
)

func myLog(args ...interface{}) {
	log.Print(append([]interface{}{"[DEBUG] "}, args))
}

func myLog2(args ...interface{}) {
	log.Print(append([]interface{}{"[DEBUG] "}, args)...)
}

func myLog3(args ...interface{}) {
	log.Print(append([]interface{}{"[DEBUG] "}, args...)...)
}

func main() {
	myLog("has ", "a ", "prefix")
	myLog2("has ", "a ", "prefix")
	myLog3("has ", "a ", "prefix")
}
````

````text
2009/11/10 23:00:00 [[DEBUG]  [has  a  prefix]]
2009/11/10 23:00:00 [DEBUG] [has  a  prefix]
2009/11/10 23:00:00 [DEBUG] has a prefix
````

Reply:
Good call!